src/main/java/org/auscope/portal/core/services/methodmakers/filter/AbstractFilter.java

```diff
@@ -602,11 +637,29 @@ private String generatePropertyComparisonFragment(String comparison, Map<String,
             sb.append(">");
         }
         sb.append(String.format("<ogc:PropertyName>%1$s</ogc:PropertyName>", propertyName));
-        if (literal != null) {
-            sb.append(String.format("<ogc:Literal>%1$s</ogc:Literal>", escapeLiteral(literal)));
+        if (function != null) {
+            sb.append(function);
+        } else {
+            if (literal != null) {

```

Comment:
Change to following:

        if (function != null) {
            sb.append(function);
        } else if (literal != null) {
            sb.append(String.format("<ogc:Literal>%1$s</ogc:Literal>", escapeLiteral(literal)));
        }